src/adapter.js

```diff
@@ -1,21 +1,84 @@
-var formatFailedStep = function(step) {
-  var stack   = step.stack;
-  var message = step.message;
+/**
+ * Decision maker for whether a stack entry is considered relevant.
+ * @param  {String}  entry Error stack entry.
+ * @return {Boolean}       True if relevant, False otherwise.
+ */
+function isRelevantStackEntry(entry) {
+  // discard empty entries:
+  return entry &&
+    // discard entries from jasmine and karma-jasmine:
+    !/node_modules\/(jasmine-core|karma-jasmine)/.test(entry) &&
+    // discard karma specifics, e.g. "at http://localhost:7018/karma.js:185"
+    !/.+\/(karma.js|context.html):.+/.test(entry);
+}
 
-  if (stack) {
-    // remove the trailing dot
-    var firstLine = stack.substring(0, stack.indexOf('\n') - 1);
+/**
+ * Returns relevant stack entries.
+ * @param  {String} stack Complete error stack trace.
+ * @return {Array}        A list of relevant stack entries.
+ */
+function getRelevantStackFrom(stack) {
+  return stack.split('\n').filter(isRelevantStackEntry);
```

Comment:
By the way I'm aware that there is an `indexOf` function (not on prototype) in the adapter.js and even tests for it. However, it is never actually used! So, unless a shim is introduced somehow, the code should have been failing on IE below 9 even before this PR.

Now, if we really want to resurrect the dinosaurs, I would recommend opening another issue specifically for this case.